index.js

```diff
@@ -79,18 +79,7 @@ function wrapFunction(config, req, res, cb) {
 
   const decryptTokenWithKeys = authentication.decryptTokenWithKeys.bind(undefined, token);
   authentication.getPublicKeys()
-    .then(decryptTokenWithKeys)
-    .then(function (decryptedToken) {
-      let validAudience = authentication.verifyAudience(decryptedToken);
-      if (!validAudience) {
-        throw new authentication.WrongAudienceError(`Token's audience ${decryptedToken.aud} did not match any for this function.`);
-      }
-      ctx.originalToken = token;
-      ctx.token = decryptedToken;
-      modules.install(ctx);
-      cb(null, ctx);
-    })
-    .catch(function (e) {
+    .then(decryptTokenWithKeysm, function (e) {
```

Comment:
Ok so did some research. When using an error handler as an argument to a promise, it will break the chain unless another value is returned, or a new Error is thrown.

the whole code block for this error handler should be
```
let message = 'Could not decrypt token with any of the public keys';
if (e instanceof authentication.WrongAudienceError) {
  message = e.message;
}
throw new Error(message);
```

Reply:
this will allow the .then and .catch blocks to run effectively. The rest of the code will still execute in the .then block

Reply:
hmmm, but we want to break it right? If there was an auth error we should stop